ambari-server/src/main/python/ambari_server/setupSso.py

```diff
@@ -113,15 +125,101 @@ def populateJwtAudiences(options, properties):
     audiences = options.sso_jwt_audience_list
 
   properties.process_pair(JWT_AUDIENCES, audiences)
+  
+def get_eligible_services(properties, admin_login, admin_password, cluster_name):
+  url = get_ambari_server_api_base(properties) + URL_TO_FETCH_SERVICES_ELIGIBLE_FOR_SSO.replace(":CLUSTER_NAME", cluster_name)
```

Comment:
`clustername` should probably be URL encoded.

Reply:
ok

Reply:
Done